ui/src/app/components/rename-modal.tsx

```diff
@@ -55,6 +54,83 @@ export class RenameModal extends React.Component<Props, States> {
     };
   }
 
+  getFileNameWithoutExtension(name: string) {
+    const lastDotIndex = name?.lastIndexOf('.');
+    return lastDotIndex === -1 ? name : name?.slice(0, lastDotIndex);
+  }
+
+  getFileExtension(name: string) {
+    const lastDotIndex = name?.lastIndexOf('.');
+    return lastDotIndex === -1 ? '' : name?.slice(lastDotIndex);
+  }
+
+  validateNewName(
+    existingNames: string[],
+    oldName: string,
+    newName: string,
+    resourceType: ResourceType
+  ) {
+    const { filteredExistingNamesNoExtension, newNameNoExtension } =
+      this.getFilteredListOfFileNamesWithoutExtension(
+        existingNames,
+        oldName,
+        newName
+      );
+
+    const errors = validate(
+      { newName: newNameNoExtension },
+      {
+        newName: nameValidationFormat(
+          filteredExistingNamesNoExtension,
+          resourceType
+        ),
+      }
+    );
+    return errors;
+  }
+
+  private getFilteredListOfFileNamesWithoutExtension(
+    existingNames: string[],
+    oldName,
+    newName: string
+  ) {
+    const oldNameExtension = this.getFileExtension(oldName);
+    const oldNameNoExtension = this.getFileNameWithoutExtension(oldName);
+
+    const newNameExtension = this.getFileExtension(newName);
+    const newNameNoExtension = this.getFileNameWithoutExtension(newName || '');
+
+    // Filtering only files that have the same extension as the original file and stripping file extensions from the names
+    const filteredExistingNamesNoExtension = existingNames
```

Comment:
I changed the approach a little bit in this iteration. Instead of checking all files, I check only the files that have the same file extension as the original file, I also strip the file extensions to make sure that the case doesn't matter.